### PR TITLE
chore: Add rollbar

### PR DESCRIPTION
**Pull request by JulesFaucherre**
[Jira](https://circleci.atlassian.net/browse/DEVEX-1138)

**To the reviewer**: would you want to merge this PR after approving - which normally should not happen - please restrain yourself

# Description

This PR adds Rollbar on the LSP.
This PR is tightly coupled to a PR on the extension that makes sure the LSP receives the information it expects: https://github.com/circleci/circleci-vscode-extension/pull/799
It is important to note that because the VSCode extension must be updated to enable Rollbar on the LSP, people won't have Rollbar enabled on their Rollbar as long as their VSCode extension is not up to date.

# Implementation details

Because the logic of the LSP is contained in the [server.commandHandler function](https://github.com/CircleCI-Public/circleci-yaml-language-server/blob/92cca700613def609af1f26096c7022f8c351d65/pkg/server/server.go#L27-L78) and that this function runs in its own goroutine, it is not possible to simply wrap the call to the `main` function with `rollbar.WrapAndWait` as [it is first proposed in the documentation](https://pkg.go.dev/github.com/rollbar/rollbar-go#hdr-Basic_Usage). Wrapping `server.commandHandler` was not possible either because we would loose the return of the function. Instead the implementation is simply, [as advised in the doc](https://pkg.go.dev/github.com/rollbar/rollbar-go#hdr-Basic_Usage), to recover the panic in the function and to pass to panic to rollbar.
After recovering and sending the panic to rollbar, we still dispatch the panic again because the panic most likely stopped the execution for a reason and it's the client's responsability to relaunch the server.

# How to validate

You can add a `panic` in the LSP code, ask me for the rollbar token, build the LSP with this token, run the extension, open a YAML, trigger the `panic` you added and make sure your panic can be seen in Rollbar

### CODE REVIEW

File: cmd/start_server/start_server.go [hunk 1]
```diff
@@ -8,9 +8,25 @@ import (
...
+	"github.com/rollbar/rollbar-go"
+)
+
+var (
+	RollbarToken = ""
 )
 
 func main() {
+	rollbar.SetToken(RollbarToken)
+	rollbar.SetEnvironment("production")
```
Comment (loderunner) [inline cmd/start_server/start_server.go:20]: Shouldn't this depend on the build? I think we should at least have:

* `development`
* `testing`
* `production`

But also, shouldn't we just disable rollbar in development and testing?

File: cmd/start_server/start_server.go [hunk 2]
```diff
@@ -8,9 +8,25 @@ import (
...
+	rollbar.SetCodeVersion(lsp.GetServerVersion())
+	rollbar.SetServerRoot("github.com/CircleCI-Public/circleci-yaml-language-server")
+
```
Comment (loderunner) [inline cmd/start_server/start_server.go:23]: I see a couple more things we could configure on Rollbar, here:

* `rollbar.SetServerHost("")`: privacy-wise, I think we shouldn't divulge the user's machine name
* `rollbar.SetPlatform('client')` and use the post_client_item token, [like recommended here for Electron,](https://docs.rollbar.com/docs/electron#node-back-end) that will be better protected against abuse

File: pkg/server/server.go [hunk 3]
```diff
@@ -27,6 +28,15 @@ type JSONRPCServer struct {
 func (server JSONRPCServer) commandHandler(_ context.Context, reply jsonrpc2.Replier, req jsonrpc2.Request) error {
 	fmt.Println("Called method: " + req.Method())
 
+	defer func() {
+		err := recover()
+		rollbar.LogPanic(err, true)
+
+		if err != nil {
+			panic(err)
+		}
```
Comment (loderunner) [inline pkg/server/server.go:37]: It seems to me we'll _always_ report an error here, even if it's `nil`. Shouldn't `LogPanic` be inside the `if` too?

```suggestion
		err := recover()

		if err != nil {
			rollbar.LogPanic(err, true)
			panic(err)
		}
```

File: pkg/server/server.go [hunk 4]
```diff
@@ -90,6 +101,9 @@ func (server JSONRPCServer) ServeStream(_ context.Context, conn jsonrpc2.Conn) e
 	}
 	conn.Go(server.ctx, server.commandHandler)
 	<-conn.Done()
+
+	rollbar.Close()
```
Comment (loderunner) [inline pkg/server/server.go:105]: Would it be better to `defer` this at the start of the function? So that whatever happens we still `Close` rollbar.

**Review by loderunner — CHANGES_REQUESTED** [review]

**Comment by JulesFaucherre** [comment]
Summary of the actual state of the PR:

A new handler has been added to the [`ExecuteCommands` methods](https://github.com/CircleCI-Public/circleci-yaml-language-server/blob/c02f16baad28ad5ca770078baf11bb0a67c96806/pkg/server/methods/executeCommand.go#L15). Method that allows for custom LSP commands. You can now run
```typescript
client.sendRequest('workspace/executeCommand', {
    command: 'setRollbarInformation',
    arguments: [
        {
           // ... any argument that must be passed as rollbar configuration
        },
    ],
});
```
With the exception of the following special values:
 - `enabled` which value will be passed to `rollbar.SetEnabled`
 - `environment` value which value will be passed to `rollbar.SetEnvironment`
 - `personId` which value will be passed as first argument of `rollbar.SetPerson`

All other properties will be passed to `rollbar.SetCustom`

**Review by loderunner — APPROVED** [review]

File: e2e/__snapshots__/definition.test.ts.snap [hunk 5]
```diff
@@ -12,11 +12,11 @@ exports[`Definition Definition on parameter 2`] = `
     "range": {
       "end": {
         "character": 28,
-        "line": 465,
+        "line": 481,
       },
       "start": {
         "character": 12,
-        "line": 462,
+        "line": 478,
```
Comment (loderunner) [inline e2e/__snapshots__/definition.test.ts.snap:19]: Why did these need updating?

Comment (JulesFaucherre) [reply e2e/__snapshots__/definition.test.ts.snap:19]: That's a very good question, it seems as the tested file uses the `mac` orb, the E2E snapshots also records the orb's diagnostics. It seems the orb has been updated and so the E2E tests also need to be updated.

What's really strange is why does the E2E tests check the `mac` orb? I haven't taken time to look into but we definitely should

**Pull request merged**